usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MealManagerClient.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+    /**
+     * Deletes all the meals of the specified pet from database.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     */
+    public void deleteAllMeals(String owner, String petName) {
+        taskManager = new TaskManager();
+        taskManager.setTaskId(DELETE);
+        taskManager.execute(BASE_URL + owner + "/" + petName);
+    }
+
+    /**
+     * Gets a meal identified by its pet and date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param date Date the meal was eaten
+     * @return The MealData identified by the data
+     */
+    public MealData getMealData(String owner, String petName, Date date) throws ExecutionException, InterruptedException {
+        taskManager = new TaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder json = taskManager.execute(BASE_URL + owner + "/" + petName + "/" + date).get();
+        return gson.fromJson(json.toString(), MealData.class);
+    }
+
+    /**
+     * Gets the data from all the specified meals from the database identified by its pet.
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @return The List containing all the meals from the pet
+     */
+    public List<Meal> getAllMealData(String owner, String petName) throws ExecutionException, InterruptedException {
+        taskManager = new TaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + "/" + petName).get();
+        String jsonArray = response.substring(1, response.length()-1);
+        String[] meals = jsonArray.split(",\\{");
+        List<Meal> mealList = new ArrayList<>();
+        mealList.add(gson.fromJson(meals[0], Meal.class));
+        for (int i=0; i < meals.length; i++) {
+            meals[i] = "{" + meals[i];
+            mealList.add(gson.fromJson(meals[i], Meal.class));
+        }
+        return mealList;
+    }
+
+    /**
+     * Gets the data from all the meals eaten by the pet between the initial and final date not including them.
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @param initialDate Initial Date
+     * @param finalDate Final Date
+     * @return The List containing all the meals eaten by the pet in the specified time
+     */
+    public List<Meal> getAllMealsBetween(String owner, String petName, String initialDate,
+                                                 String finalDate) throws ExecutionException, InterruptedException {
+        taskManager = new TaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response =
+            taskManager.execute(BASE_URL + owner + "/" + petName + "/" + initialDate + "/" + finalDate).get();
+        String jsonArray = response.substring(1, response.length()-1);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['-' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)